homework.rb

```diff
@@ -1,6 +1,15 @@
-VARIANT = 0
+VARIANT = 12
+NUM_OF_LETTERS = 6
+NUM_OF_DIGITS = 8
+DIGITS_SPLIT_PLACE = 6
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  # pattern: "ABCDEF-123456-98"
+  # chunk of additional code to add (hopefully) more functionality:
+  temp_str = ''
+  NUM_OF_DIGITS.times { temp_str += rand(0...10).to_s }
+  # the main code:
+  ('a'..'z').to_a.sample(NUM_OF_LETTERS).join.upcase + '-' +
```

Comment:
The most obvious one is using backslashes at the end of each part of expression, if it is what you've meant.